packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -431,6 +457,41 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     }
   }
 
+  /// Checks whether an enrollment with the same appName and deviceName already exists for the given request.
+  /// If a matching enrollment is found, [AtEnrollmentException] exception is thrown.
+  /// Otherwise, the enrollment request is accepted.
+  @visibleForTesting
+  Future<void> validateEnrollmentRequest(EnrollParams enrollParams) async {
+    // Fetches all the enrollment keys from the keystore.
+    List<dynamic> enrollmentKeys = keyStore.getKeys(regex: 'enrollments');
+
+    // Iterate all the enrollment keys and verify if there exists an enrollment the same appName and deviceName combination
```

Comment:
Iterate through the existing enrollments and verify that there is no enrollment with the same appName and deviceName combination, and a status of 'pending' or 'approved'